flex/validation/common.py

```diff
@@ -390,6 +378,8 @@ def validate_object(obj, field_validators=None, non_field_validators=None,
 
     if 'discriminator' in schema:
         schema_validators = add_polymorphism_requirements(obj, schema, context, schema_validators)
+        # delete resolved discriminator to avoid infinite recursion
+        del schema['discriminator']
```

Comment:
Given that this mutates `schema` can you add a `copy.deepcopy` to the entry points in https://github.com/pipermerriam/flex/blob/master/flex/core.py to make a copy of the schema so that we're *sure* that we aren't modifying a *master* schema of some sort and that this mutation is isolated.

Reply:
I've added a deepcopy